### PR TITLE
DG: mailing list improvements

### PR DESCRIPTION
**Pull request by cookiedan42**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -403,21 +403,30 @@ Step 7. CONNECTIONS `UI` will observe a change in the `ObservableList<Person>` a
...
-Step 1. The user filters the contacts using other commands, eg. `find`.
+Step 1. The user filters the contacts using other commands, eg. `find`.  
+Step 2. The `FilteredList` in `Model` is updated  
+Step 3. The UI is updated to reflect this new state.  
+Step 4. The user provides a series of prefixes to `mailingList` to pick the fields. If no arguments are provided, default selectors are used.  
+Step 5. These `Prefix` arguments are stored in `Model`  
+Step 6. The user is prompted to pick the name and download location of their generated CSV file.  
+Step 7. The `FilteredList`, `Prefixes` and `Path` are passed to `CsvUtil#modelToCsv`, which will serialize and write the CSV file.  
+![MailingListSequenceDiagram](images/MailingListSequenceDiagram.png)  
+Step 8. The header row is created based on `Prefix` arguments stored in `Model`, based on a mapping in `CsvUtil`
```
Comment (yucheng11122017) [inline docs/DeveloperGuide.md:420]: I think you are missing a full stop here :)

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -403,21 +403,30 @@ Step 7. CONNECTIONS `UI` will observe a change in the `ObservableList<Person>` a
...
+Step 2. The `FilteredList` in `Model` is updated  
+Step 3. The UI is updated to reflect this new state.  
+Step 4. The user provides a series of prefixes to `mailingList` to pick the fields. If no arguments are provided, default selectors are used.  
+Step 5. These `Prefix` arguments are stored in `Model`  
+Step 6. The user is prompted to pick the name and download location of their generated CSV file.  
+Step 7. The `FilteredList`, `Prefixes` and `Path` are passed to `CsvUtil#modelToCsv`, which will serialize and write the CSV file.  
+![MailingListSequenceDiagram](images/MailingListSequenceDiagram.png)  
+Step 8. The header row is created based on `Prefix` arguments stored in `Model`, based on a mapping in `CsvUtil`  
+Step 9. Individual rows are generated based on the `Prefix` arguments stored in `Model` and the `FilteredPerson` in `ModelManager`, based on a mapping in `CsvUtil`  
+Step 10. The headers and rows are written to the CSV file that is specified by the user
```
Comment (yucheng11122017) [inline docs/DeveloperGuide.md:422]: Missing full stop here also :)

**Review by yucheng11122017 — APPROVED** [review]
LGTM except for some minor stuff :)

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -403,21 +403,30 @@ Step 7. CONNECTIONS `UI` will observe a change in the `ObservableList<Person>` a
...
 
-Step 1. The user filters the contacts using other commands, eg. `find`.
+Step 1. The user filters the contacts using other commands, eg. `find`.  
+Step 2. The `FilteredList` in `Model` is updated  
+Step 3. The UI is updated to reflect this new state.  
+Step 4. The user provides a series of prefixes to `mailingList` to pick the fields. If no arguments are provided, default selectors are used.  
+Step 5. These `Prefix` arguments are stored in `Model`  
+Step 6. The user is prompted to pick the name and download location of their generated CSV file.  
+Step 7. The `FilteredList`, `Prefixes` and `Path` are passed to `CsvUtil#modelToCsv`, which will serialize and write the CSV file.  
+![MailingListSequenceDiagram](images/MailingListSequenceDiagram.png)
```
Comment (Justinhoejj) [inline docs/DeveloperGuide.md:419]: lines 2,5,7 no fullstop

**Review by Justinhoejj — APPROVED** [review]
LGTM except for punctuation issues.

**Pull request merged**